### PR TITLE
feat(ai): add @kaiord/ai package for LLM workout parsing

### PR DESCRIPTION
**Pull request by pablo-albaladejo**
## Summary

- Add new `@kaiord/ai` package that converts natural language workout descriptions into structured KRD `Workout` objects using Vercel AI SDK v6
- `createTextToWorkout()` factory accepts a provider-agnostic `LanguageModel` — consumer chooses Anthropic, OpenAI, etc.
- Comprehensive system prompt with multi-language support (ES/EN), training abbreviation glossary, and pace/power conversion rules
- Retry loop with validation error feedback, input sanitization, post-processing (stepIndex re-indexing, name override)
- CI/CD: added to changeset config, ci.yml test matrix, changeset-bot, release pipeline, and GitHub releases script

## Architecture

```
NL text → createTextToWorkout(model) → Workout → createWorkoutKRD() → KRD → garminWriter → GCN
```

Foundation for a future `@kaiord/aws-agent` serverless package (Lambda + API GW + Telegram Bot).

## Test plan

- [x] 24 unit tests passing (mocked AI SDK)
- [x] 98%+ code coverage
- [x] Full monorepo build clean
- [x] Lint zero warnings across all packages
- [x] TypeScript strict mode, no errors
- [ ] Manual E2E with `ANTHROPIC_API_KEY` (not in CI)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a new AI package: LLM-powered natural-language → workout parser with input validation, step reindexing, name override, structured prompts, and a dedicated AI parsing error type.
* **Tests**
  * Expanded unit tests covering parsing flows, retries, input validation, prompt loading, reindexing, and error behavior.
* **Documentation**
  * New prompt, skill, agent, and reference docs for workout parsing, GCN output, zones, and tooling guidance.
* **Chores**
  * CI, release, build, tooling, and audit configs updated to build, test, and publish the new AI package; added size/quality tooling configs.

### CODE REVIEW

File: packages/ai/src/adapters/validate-input.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    throw createAiParsingError("Input text is empty", text, 0);
+  }
+
+  if (sanitized.length > MAX_INPUT_LENGTH) {
+    throw createAiParsingError(
+      `Input text exceeds ${MAX_INPUT_LENGTH} characters (got ${sanitized.length})`,
+      text.slice(0, 100),
+      0
+    );
+  }
```
[the remark this answers is not in the document]
Comment (pablo-albaladejo) [reply packages/ai/src/adapters/validate-input.ts]: ✅ **FIXED** in commit 58d0b6f9

Changed `text.slice(0, 100)` to `text` — the full original input is now stored in `AiParsingError.inputText`. Consumers can truncate for display if needed.

File: packages/ai/src/prompts/parse-workout.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  "duration": { "type": "time", "seconds": 600 },
+  "targetType": "pace",
+  "target": { "type": "pace", "value": { "unit": "mps", "value": 3.33 } },
+  "intensity": "warmup",
+  "notes": "optional note"
+}
+```
+
+Required fields: `stepIndex`, `durationType`, `duration`, `targetType`, `target`.
+Optional: `intensity`, `notes`, `equipment`.
```
[the remark this answers is not in the document]
Comment (pablo-albaladejo) [reply packages/ai/src/prompts/parse-workout.md]: ✅ **FIXED** in commit 58d0b6f9

Removed `equipment` from the optional fields list. The LLM has no guidance on valid values, so it's better to omit it until the prompt explicitly documents the schema.

**Comment by pablo-albaladejo** [comment]
@coderabbitai review

**Comment by pablo-albaladejo** [comment]
## 🔬 Deep Multi-Agent Review — All Findings Addressed

4 specialized agents (code-reviewer, arch-guardian, test-analyst, cicd-guardian) performed an exhaustive review from 7 perspectives. All findings fixed in commit \`53ea4e40\`.

### Critical (2/2 fixed)

| # | Finding | Fix |
|---|---------|-----|
| 1 | **release.yml** missing @kaiord/ai version tracking, build step, and summary | Added AI_VERSION_BEFORE/AFTER, build step, ai-changed/ai-version outputs, summary entry |
| 2 | **AiParsingError.inputText** stored raw unsanitized user text (PII risk) | Truncated to 200 chars max, using sanitized text |

### Important (6/6 fixed)

| # | Finding | Fix |
|---|---------|-----|
| 3 | **changeset-bot.yml** PR comment missing @kaiord/ai | Added to instructional package list |
| 4 | System prompt **notes field** could leak instructions | Added: "notes must ONLY contain user input" |
| 5 | **reindexSteps** gap: RepetitionBlocks consumed a stepIndex | Used counter that skips blocks |
| 6 | **Retry prompt** lastError not truncated (injection risk) | Truncated to 200 chars |
| 7 | **Function size** 54 lines > 40 limit | Extracted `executeWithRetry` helper |
| 8 | **Logger** `?.debug?.()` double optional chaining | Fixed to `?.debug()` (port guarantees methods) |

### Suggestions (4/5 addressed)

| # | Finding | Status |
|---|---------|--------|
| 9 | Missing test for invalid sport option | ✅ Added |
| 10 | Duplicate sourceMappingURL in dist | ✅ Investigated — monorepo-wide tsup behavior, not a regression |
| 11 | No README.md | 📝 Deferred to follow-up (not blocking) |
| 12 | errors.ts has no dedicated test file | ✅ Added `errors.test.ts` |
| 13 | Missing port type in core | 📝 Deferred — package is intentionally flat for v1 |

### Test Results ✅

- **37 tests** passing (was 24, +13 new)
- **5 test files** (was 4, +1 new `errors.test.ts`)
- **Coverage thresholds** enforced: 80% lines/functions/branches/statements
- All edge cases covered: empty input, max length, invalid sport, null output, retry exhaustion, non-Error thrown, control-only input, config defaults, mixed step types

### Architecture ✅
- Hexagonal compliance: depends only on `@kaiord/core`, `ai` as peer dep
- File sizes: all ≤ 100 lines, functions ≤ 40 lines
- Type-only imports correctly separated
- Factory pattern matches other adapters

### CI/CD ✅
- release.yml: full version tracking + build + summary
- ci.yml: detect-changes + test matrix + build verification
- changeset-bot: PUBLISHABLE + comment text
- changeset config: linked array

🤖 Generated with [Claude Code](https://claude.com/claude-code)

**Comment by pablo-albaladejo** [comment]
@coderabbitai review

File: packages/ai/src/adapters/reindex-steps.test.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+              targetType: "open",
+              target: { type: "open" },
+            },
+          ],
+        },
+      ],
+    };
+
+    const result = reindexSteps(workout);
+    const block = result.steps[0];
```
[the remark this answers is not in the document]
Comment (pablo-albaladejo) [reply packages/ai/src/adapters/reindex-steps.test.ts]: ✅ **FIXED** in commit b39ff1d8

Added blank line between Act (`reindexSteps(workout)`) and Assert (`const block = result.steps[0]`) to satisfy the AAA pattern.

File: .claude/agents/bundle-optimizer.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+| Package | Before | After | Threshold | Status |
+|---------|--------|-------|-----------|--------|
+| core    | X KB   | Y KB  | 50 KB     | OK/OVER|
+
+- Optimizations applied: N
+- Total size reduction: X KB -> Y KB
+- Tests: PASS/FAIL
+- Build: PASS/FAIL
+- Lint: PASS/FAIL
+```
```
[the remark this answers is not in the document]
Comment (pablo-albaladejo) [reply .claude/agents/bundle-optimizer.md]: ✅ **FIXED** in commit b39ff1d8

Added `text` language tag to the output template code fence.

File: .dependency-cruiser.cjs [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+      comment:
+        "Format adapter packages (fit, tcx, zwo, garmin) must only depend on @kaiord/core",
+      severity: "error",
+      from: {
+        path: "packages/(fit|tcx|zwo|garmin)/src",
+      },
+      to: {
+        path: "packages/(cli|mcp|garmin-connect|ai|workout-spa-editor)/src",
+      },
+    },
```
[the remark this answers is not in the document]
Comment (pablo-albaladejo) [reply .dependency-cruiser.cjs]: ✅ **FIXED** in commit b39ff1d8

Updated the `adapter-depends-only-on-core` rule to also forbid cross-imports between adapter packages (fit, tcx, zwo, garmin). The `to.path` now includes all adapter packages in addition to the other forbidden targets.
